src/controllers/midi/midicontroller.cpp

```diff
@@ -288,7 +288,7 @@ void MidiController::processInputMapping(const MidiInputMapping& mapping,
             return;
         }
 
-        QScriptValue function = pEngine->resolveFunction(mapping.control.item);
+        QScriptValue function = pEngine->wrapFunctionCode(mapping.control.item, 5);
```

Comment:
OK, the script value holds a pointer of the script engine and vice versa. So it can be dangerous to spread QStringValues around. 

How about turn the current cache solution into a shared pointer cache. The mapping can hold a weak pointer. 
Turing a weak pointer into a shared pointer and execute the QScriptValue function should be much faster then looking up a hash. If the strong cache is gone, all stored weak pointers are invalid and the mapping has to evaluate its functions again. 

```
QHash<QString, QSharedPointer<QScriptValue> > m_scriptWrappedFunctionCache
...
QSharedPointer<QScriptValue> = pFunction(new QScriptValue(m_pEngine->evaluate(codeSnippet)));
m_scriptWrappedFunctionCache[codeSnippet] = pFunction; 
...

...
QWeakPointer<QScriptValue> scriptfunction;
...
mapping.scriptfunction = pFunction; 
...

QSharedPointer<QScriptValue> strong = mapping.scriptfunction.toStrongRef();
if (strong)
    // execute
else
   // evaluate again 
```